worlds/zillion/__init__.py

```diff
@@ -242,6 +242,42 @@ def generate_basic(self) -> None:
         self.world.completion_condition[self.player] = \
             lambda state: state.has("Win", self.player)
 
+    @staticmethod
+    def stage_generate_basic(multiworld: MultiWorld, *args: Any) -> None:
+        # item link pools are about to be created in main
+        # JJ can't be an item link unless all the players share the same start_char
+        # (The reason for this is that the JJ ZillionItem will have a different ZzItem depending
+        #  on whether the start char is Apple or Champ, and the logic depends on that ZzItem.)
+        for group in multiworld.groups.values():
+            # TODO: remove asserts on group when we can specify which members of TypedDict are optional
+            assert "game" in group
+            if group["game"] == "Zillion":
+                assert "item_pool" in group
+                item_pool = group["item_pool"]
+                to_stay = "JJ"
+                if "JJ" in item_pool:
+                    assert "players" in group
+                    group_players = group["players"]
+                    start_chars = cast(Dict[int, ZillionStartChar], getattr(multiworld, "start_char"))
+                    players_start_chars = [
+                        (player, start_chars[player].get_current_option_name())
+                        for player in group_players
+                    ]
+                    start_char_counts = Counter(sc for _, sc in players_start_chars)
+                    # majority rules
+                    if start_char_counts["Apple"] > start_char_counts["Champ"]:
+                        to_stay = "Apple"
+                    elif start_char_counts["Champ"] > start_char_counts["Apple"]:
+                        to_stay = "Champ"
+                    else:  # equal
+                        to_stay = multiworld.random.choice(("Apple", "Champ"))
+
+                    for p, sc in players_start_chars:
+                        if sc != to_stay:
+                            group_players.remove(p)
```

Comment:
wouldn't it make more sense to split the group in two?

Reply:
I don't feel like I know the item_link system well enough yet to be confident trying to do that.
(I don't know if this is a safe place to make a group. I don't know what processing the groups have already gone through. etc.)
I'll look at it some more.

Reply:
After thinking about it more, I think the added complexity of splitting the group is not worth it.
I think it will fit with players' intuition that rescue item links will only work if the players start with the same char, so they can set them up that way.

What might be nice for this is "option links", so a group of players can have the "random" (or "mystery") choose the same option for everyone in the group.